tools/resource-gen.go

```diff
@@ -204,13 +137,26 @@ func (l *{{.ResourceName}}List) GetPagination() *model.Pagination {
 	return &l.Pagination
 }
 
-{{if not .SkipRegistration}}
+var {{.ResourceName}}TypeDescriptor model.ResourceTypeDescriptor 
```

Comment:
Did you consider making `Descriptor()` part of the Resource interface, similar to `Scope()`? I think that would be consistent and convenient, though I'm not sure it would completely let us get rid of the globals.

Reply:
I did though it doesn't really solve the _globals_ because we don't always have a `Resource` as input.